### PR TITLE
Moved tests out of contrib.

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

**Comment by aaugustin** [comment]
I can't say I reviewed every line of every commit :-) but the changes amount to a solid cleanup. I didn't spot anything that would need fixing. Please merge before the PR becomes stale.

File: tests/auth_tests/test_basic.py [hunk 1]
```diff
@@ -4,7 +4,6 @@
 from django.contrib.auth import get_user_model
 from django.contrib.auth.models import AnonymousUser, User
 from django.contrib.auth.tests.custom_user import CustomUser
```
Comment (MarkusH) [inline tests/auth_tests/test_basic.py]: Any reason why you didn't move that?

Comment (timgraham) [inline tests/auth_tests/test_basic.py]: Yes, it contains some documented stuff. https://docs.djangoproject.com/en/1.7/topics/auth/customizing/#custom-users-and-testing-fixtures I could try to move the undocumented stuff at least, but was wondering if maybe the module could be deprecated entirely.

I think `skipIfCustomUser` should at least be pretty safe to deprecate with this change.
https://code.djangoproject.com/ticket/22993

Comment (MarkusH) [inline tests/auth_tests/test_basic.py]: Thanks, I was looking for that.

Comment (timgraham) [inline tests/auth_tests/test_basic.py]: I added a commit to move the non-documented auth models to the test package. It's a nice cleanup.

**Pull request merged**